test/WebJobs.Script.Tests/Binding/SendGridScriptBindingProviderTests.cs

```diff
@@ -1,129 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-#if ALLEXTENSIONS
-using System;
-using System.Diagnostics;
-using System.Reflection;
-using Microsoft.Azure.WebJobs.Extensions.SendGrid;
-using Microsoft.Azure.WebJobs.Host.Config;
-using Microsoft.Azure.WebJobs.Script;
-using Microsoft.Azure.WebJobs.Script.Binding;
-using Microsoft.Azure.WebJobs.Script.Extensibility;
-using Newtonsoft.Json.Linq;
-using Xunit;
-
-namespace Microsoft.Azure.WebJobs.Script.Tests
-{
-    public class SendGridScriptBindingProviderTests
-    {
-        private readonly ScriptBindingProvider _provider;
-
-        public SendGridScriptBindingProviderTests()
-        {
-            JobHostConfiguration config = new JobHostConfiguration();
-            config.AddExtension(new SendGridConfiguration());
-            TestTraceWriter traceWriter = new TestTraceWriter(TraceLevel.Verbose);
-            JObject hostMetadata = new JObject();
-
-            var provider = new GeneralScriptBindingProvider(config, hostMetadata, traceWriter);
-            var metadataProvider = new JobHost(config).CreateMetadataProvider();
-            provider.CompleteInitialization(metadataProvider);
-            _provider = provider;
-        }
-
-        private static SendGridConfiguration CreateConfiguration(JObject config)
-        {
-            var ctx = new ExtensionConfigContext
-            {
-                Config = new JobHostConfiguration()
-                {
-                    HostConfigMetadata = config
-                },
-                Trace = new TestTraceWriter(TraceLevel.Verbose)
-            };
-            SendGridConfiguration result = new SendGridConfiguration();
-            result.Initialize(ctx);
-
-            return result;
-        }
-
-        [Fact]
-        public void CreateConfiguration_CreatesExpectedConfiguration()
-        {
-            JObject config = new JObject();
-
-            var result = CreateConfiguration(config);
-
-            Assert.Null(result.FromAddress);
-            Assert.Null(result.ToAddress);
-
-            config = new JObject
```

Comment:
Ideally these unit tests would be moved to the unit test suites for the extensions. Since these are official bindings, we need to make sure we retain the tests for config.

I saw the e2e test issue you created, but these are really unit tests, and they'd live with the extension.

Reply:
Fabio had some opinions we shouldn't have script related tests in extensions. You feel differently?

Reply:
From offline discussion, for host configuration type tests, I'll send PRs moving them to extensions. For function.json parsing tests, we'll have tests in the e2e suite (plan still kinda TBD, though)

Reply:
Configuration tests already exist in SendGrid tests: https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/test/WebJobs.Extensions.Tests/Extensions/SendGrid/SendGridHelpersTests.cs#L146